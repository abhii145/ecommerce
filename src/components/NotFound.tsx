import { Link } from "react-router"
import banner from "../assets/404.svg"

const NotFound = () => {
  return (
    <div
      className="flex flex-col items-center justify-center min-h-screen"
      role="main"
    >
      <img
        src={banner}
        alt="404 Not Found"
        className="w-1/2 h-auto"
        loading="lazy"
      />
      <h1 className="text-4xl font-bold mb-2">Page Not Found</h1>
      <p className="text-lg mb-4">
        Sorry, the page you are looking for does not exist.
      </p>
      <Link
        to="/"
        className="text-blue-500 underline"
        aria-label="Go back to Home"
      >
        Go back to Home
      </Link>
    </div>
  )
}

export default NotFound
