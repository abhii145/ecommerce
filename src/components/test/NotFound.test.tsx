import React from "react"
import { render } from "@testing-library/react"
import { BrowserRouter as Router } from "react-router"
import NotFound from "../NotFound"

describe("NotFound Component", () => {
  it("matches snapshot", () => {
    const { asFragment } = render(
      <Router>
        <NotFound />
      </Router>
    )
    expect(asFragment()).toMatchSnapshot()
  })

  it("renders NotFound component", () => {
    const { getByText } = render(
      <Router>
        <NotFound />
      </Router>
    )
    expect(getByText(/page not found/i)).toBeInTheDocument()
    expect(getByText(/go back to home/i)).toBeInTheDocument()
  })
})
