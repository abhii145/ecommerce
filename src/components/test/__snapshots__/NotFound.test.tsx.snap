// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFound Component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="flex flex-col items-center justify-center min-h-screen"
    role="main"
  >
    <img
      alt="404 Not Found"
      class="w-1/2 h-auto"
      loading="lazy"
      src="test-file-stub"
    />
    <h1
      class="text-4xl font-bold mb-2"
    >
      Page Not Found
    </h1>
    <p
      class="text-lg mb-4"
    >
      Sorry, the page you are looking for does not exist.
    </p>
    <a
      aria-label="Go back to Home"
      class="text-blue-500 underline"
      data-discover="true"
      href="/"
    >
      Go back to Home
    </a>
  </div>
</DocumentFragment>
`;
